lib/tests/streamlit/runtime/scriptrunner/script_runner_test.py

```diff
@@ -375,17 +414,15 @@ def test_FragmentStorageKeyError_becomes_RuntimeError(
     @patch("streamlit.runtime.scriptrunner.exec_code.handle_uncaught_app_exception")
     def test_regular_KeyError_is_rethrown(self, patched_handle_exception):
         fragment = MagicMock()
-        fragment.side_effect = KeyError("kaboom")
-
         scriptrunner = TestScriptRunner("good_script.py")
-        scriptrunner._fragment_storage.set("my_fragment", fragment)
+        scriptrunner._fragment_storage.set("my_fragment_", fragment)
 
         scriptrunner.request_rerun(RerunData(fragment_id_queue=["my_fragment"]))
         scriptrunner.start()
         scriptrunner.join()
 
         ex = patched_handle_exception.call_args[0][0]
-        assert isinstance(ex, KeyError)
+        assert isinstance(ex, (RuntimeError, KeyError))
```

Comment:
Hm, the purpose of this test is to ensure that we don't incorrectly convert `KeyError`s to `FragmentStorageKeyError`s like we previously did (the `except` handler for fragments not being found was previously written too broadly so that it would catch any `KeyError` instance), so I wouldn't expect this to have to be changed.

If changing this was necessary in order to get it to pass, I think something weird must be going on here.

Reply:
Oh yeah. So the behavior was still correct (I checked it manually), but due to the refactoring the mocked fragment was not correct anymore and then I interpreted and changed the test wrongly. I have updated the test now to check the original intent!